capsules/src/led.rs

```diff
@@ -7,20 +7,20 @@ use kernel::{AppId, Driver, ReturnCode};
 use kernel::hil;
 
 /// Whether the LEDs are active high or active low on this platform.
+#[derive(Clone,Copy)]
 pub enum ActivationMode {
     ActiveHigh,
     ActiveLow,
 }
 
 pub struct LED<'a, G: hil::gpio::Pin + 'a> {
-    pins: &'a [&'a G],
-    mode: ActivationMode,
+    pins_init: &'a [(&'a G, ActivationMode)],
```

Comment:
This could probably benefit from being it's own struct, instead of a tuple, that implements `on`, `off`, and `toggle` and the syscall driver would be a thing wrapper around that. That way that struct would itself be a capsule usable by other apps/capsules in the kernel.